dtls/client-dtls-nonblocking.c

```diff
@@ -202,61 +104,111 @@ int main (int argc, char** argv)
     wolfSSL_set_fd(ssl, sockfd);
     wolfSSL_set_using_nonblock(ssl, 1);
     fcntl(sockfd, F_SETFL, O_NONBLOCK);
-    NonBlockingDTLS_Connect(ssl);
 
-    DatagramClient(ssl);
-    while ( (wolfSSL_write(ssl, srTest, sizeof(srTest))) != sizeof(srTest)) {
-	    printf("failed to write");
-        return 1;
-    }
+/*****************************************************************************/
+/*                     Non-blocking code for DTLS connect                    */
 
-    session = wolfSSL_get_session(ssl);
-    sslResume = wolfSSL_new(ctx);
+    int ret = wolfSSL_connect(ssl);
```

Comment:
Please move all variable declarations to the top of the function.  Some compilers will throw warnings about this.